File: modify_category.js
const fs = require('fs');
const path = require('path');

const filepath = path.join('c:', 'eclub', 'pc', 'pages', 'category.html');
const content = fs.readFileSync(filepath, 'utf-8');

// The file might be broken right now. Let's fix it manually.
// First, find <div class="product-list">
// Then find <!-- 하단 페이지네이션 구조 --> (if exists) or <div class="pagination">

const listRegex = /(<div\s+class="product-list">)([\s\S]*?)(<div\s+class="pagination">)/;
const match = listRegex.exec(content);

if (!match) {
    console.error("Could not find product-list block");
    process.exit(1);
}

const prefix = content.slice(0, match.index + match[1].length);
const itemsHtml = match[2];
const suffix = content.slice(match.index + match[1].length + match[2].length);

const itemRegex = /(<!--\s*Product Item.*?\n)?(\s*<div class="product-item"[\s\S]*?<\/div>\s*<\/div>\s*<\/div>)/gs;
let itemMatches = [...itemsHtml.matchAll(itemRegex)];

if (itemMatches.length < 4) {
    console.error(`Found only ${itemMatches.length} items. Expected at least 4.`);
    process.exit(1);
}

let newItemsHtml = "";

for (let i = 0; i < 30; i++) {
    const templateMatch = itemMatches[i % 4];
    let templateBody = templateMatch[2];
    
    // Generate random sort variables
    const sales = Math.floor(Math.random() * 990) + 10;
    const purchases = Math.floor(Math.random() * 495) + 5;
    const isNew = Math.floor(Math.random() * 100);
    const priceBase = (Math.floor(Math.random() * 490) + 10) * 100;
    
    // Replace ranking number
    const rankNum = (i + 1).toString().padStart(2, '0');
    templateBody = templateBody.replace(/(<span class="badge-ranking">)\d+위(<\/span>)/g, `$1${rankNum}위$2`);
    
    // Replace price display
    const formattedPrice = priceBase.toLocaleString('ko-KR');
    templateBody = templateBody.replace(/(<div class="main-price">[\s\S]*?<span class="price">)[0-9,]+원(<\/span>)/g, `$1${formattedPrice}원$2`);
    
    // Add sorting attributes
    const dataAttrs = ` data-sales="${sales}" data-purchase="${purchases}" data-new="${isNew}" data-price="${priceBase}"`;
    if (templateBody.includes('data-delivery-type')) {
        templateBody = templateBody.replace(/data-sales="\d+"\s+data-purchase="\d+"\s+data-new="\d+"\s+data-price="\d+"\s+/, '');
        templateBody = templateBody.replace('data-delivery-type', dataAttrs.trim() + ' data-delivery-type');
    } else {
        templateBody = templateBody.replace('class="product-item"', `class="product-item"${dataAttrs}`);
    }
    
    newItemsHtml += `\n                        <!-- Product Item ${i+1} -->${templateBody}`;
}

const fixedEnd = `
                    </div>

                    <!-- 하단 페이지네이션 구조 -->
                    `;

const paginationStart = `<div class="pagination">
                        <button type="button" class="btn-page-first">«</button>
                        <button type="button" class="btn-page-prev">‹</button>
                        <button type="button" class="active">1</button>
                        <button type="button">2</button>
                        <button type="button">3</button>
`;

let finalSuffix = suffix;
// If the suffix from the current file misses the buttons because of the bad replace:
if (!finalSuffix.includes('btn-page-first')) {
    finalSuffix = finalSuffix.replace('<div class="pagination">', paginationStart);
}

const newContent = prefix + newItemsHtml + fixedEnd + finalSuffix;
fs.writeFileSync(filepath, newContent, 'utf-8');
console.log("Successfully generated 30 product items with fixed layout via Node.");

File: modify_category.py
import re
import random

def main():
    filepath = r'c:\eclub\pc\pages\category.html'
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find the block where <div class="product-list">...</div> is defined
    # We will grab all items between <div class="product-list"> and <!-- 하단 페이지네이션 구조 -->
    match = re.search(r'(<div\s+class="product-list">)([\s\S]*?)(<div\s+class="pagination">)', content)
    if not match:
        print("Could not find product-list block")
        return

    prefix = content[:match.end(1)]
    items_html = match.group(2)
    suffix = match.group(3) + content[match.end(3):]

    # Extract the 4 actual product items
    # They are separated by HTML comments. Let's just find all <div class="product-item"...>...</div>
    # Using a simple block extractor
    item_pattern = re.compile(r'(<!--\s*Product Item.*?\n)?(\s*<div class="product-item".*?</div>\s*</div>\s*</div>)', re.DOTALL)
    items = item_pattern.findall(items_html)
    
    if len(items) < 4:
        print(f"Found only {len(items)} items. Expected 4.")
        return

    new_items_html = ""
    for i in range(30):
        # We grab the actual html element from the tuple match
        template_comment, template_body = items[i % 4]
        
        # Modify the template body
        sales = random.randint(10, 1000)
        purchases = random.randint(5, 500)
        is_new = random.randint(0, 100) # higher is newer score
        price_base = random.randint(10, 500) * 100
        
        # change ranking number (e.g. 01위)
        template_body = re.sub(r'(<span class="badge-ranking">)\d+위(</span>)', rf'\g<1>{i+1:02d}위\g<2>', template_body)
        
        # change price (e.g. 36,800원) - only the main price
        template_body = re.sub(r'(<div class="main-price">[\s\S]*?<span class="price">)[0-9,]+원(</span>)', 
                               rf'\g<1>{price_base:,}원\g<2>', template_body)
        
        # Add sorting data attributes to the product-item
        data_attrs = f' data-sales="{sales}" data-purchase="{purchases}" data-new="{is_new}" data-price="{price_base}"'
        if 'data-delivery-type' in template_body:
            template_body = template_body.replace('data-delivery-type', data_attrs + ' data-delivery-type', 1)
        else:
            template_body = template_body.replace('class="product-item"', f'class="product-item"{data_attrs}', 1)
            
        new_items_html += f'\n                        <!-- Product Item {i+1} -->{template_body}'

    new_content = prefix + new_items_html + '\n                    ' + suffix
    
    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("Successfully generated 30 product items with random sort data.")

if __name__ == '__main__':
    main()
